packages/neon/neon/lib/src/blocs/timer.dart

```diff
@@ -0,0 +1,72 @@
+part of '../../neon.dart';
+
+abstract class TimerBlocEvents {
+  /// Register a [callback] that will be called periodically.
+  /// The time between the executions is defined by the [duration].
+  TimerBlocTimer registerTimer(final Duration duration, final VoidCallback callback);
+
+  /// Unregister a timer that has been previously registered with the bloc.
+  /// You can also use [TimerBlocTimer.cancel].
+  void unregisterTimer(final TimerBlocTimer timer);
+}
+
+abstract class TimerBlocStates {}
+
+/// Execute callbacks at defined periodic intervals.
+/// Components can register their callbacks and everything with the same periodicity will be executed at the same time.
+///
+/// The [TimerBloc] is a singleton.
+/// Sub-second timers are not supported.
+class TimerBloc extends Bloc implements TimerBlocEvents, TimerBlocStates {
+  factory TimerBloc() => _singleton;
+  TimerBloc._();
+
+  static final TimerBloc _singleton = TimerBloc._();
+
+  final Map<int, Timer> _timers = {};
+  final Map<int, List<VoidCallback>> _callbacks = {};
```

Comment:
can we make it a `Set<VoidCallback>`?

I don't think it makes sense to add a callback twice.

Reply:
Hm you could have a callback with multiple intervals? Although that seems quite useless so we might just not support it

Reply:
A callback with multiple intervals would place the callback in different sets.

I meant `Map<int, Set<VoidCallback>>` still having one set per intervall.